src/printer.js

```diff
@@ -2892,6 +2892,43 @@ function shouldGroupFirstArg(args) {
 
 function printArgumentsList(path, options, print) {
   const printed = path.map(print, "arguments");
```

Comment:
All your code looks really complicated, I would change this block. If the argument is not the last one and there's an empty line after, then instead of printed be the printed element, it's `concat([printedArg, hardline])`